docs/en/Modal.md

```diff
@@ -0,0 +1,57 @@
+# `<Modal>`
+
+Creates a modal overlay. This component is similar to [`<Overlay>`](./Overlay.md) but provides the following extras:
+
+- traps tabbing focus inside the overlay.
+- makes all other root level elements inert to user input.
+- sets aria title and description ids.
+
+### Usage
+
+Basic example.
+
+```jsx
+import {Modal} from 'libreact/lib/Modal';
+
+<Modal>
+  This is modal.
+</Modal>
+```
+
+Set aria title and description.
+
+```jsx
+<Modal>{({idTitle, idDescription}) =>
+  <div>
+    <h1 id={idTitle}>My title</h1>
+    <p id={idDescription}>This is description.</p>
+  </div>
+}</Modal>
+```
+
+Track when user intends to close the the modal.
+
+```jsx
+<Modal onClick={} onEsc={}>
```

Comment:
instead of `onEsc` wouldn't `onClose` be more generic?

Reply:
`onClose` is indeed more generic, but here I want to explicitly provide `Esc` key event, so user can decided for herself what she wants to do with `Esc` press event.

Reply:
the fact it gets triggered with Esc it is a technical detail. On mobile there won't be an esc key event right?

Reply:
Technical detail, but important.

Reply:
"for herself"

Reply:
Actually its not technical, its semantic detail.